map type declaration does not support imported module's enum as key type

<!-- Please make sure to run `v up` before reporting any issues as it may have already been fixed.
     It's also advisable to update all relevant modules using `v outdated` and `v install` -->

<!-- You can use `v doctor` to fill up the next fields -->
**V version:** V 0.2.2 dd0a975

**OS:** windows, Microsoft Windows 10 Enterprise v19042 64-Bit

**What did you do?**

Used an enum type from an imported module as a key type in the declaration of a map type.

Run `v -cg test .` inside a folder with the following files:

```v
//m.v
module m

pub enum MyEnum {
	one
	two
}

pub type MyEnumMap = map[MyEnum]string

pub struct MyStruct {
pub:
	field map[MyEnum]string
}
```

```v
//m_literal_test.v
import m

const literal_map = map{
	m.MyEnum.one: "one"
	m.MyEnum.two: "two"
}

fn test_literal() {
	assert literal_map[m.MyEnum.one] == "one"
}
```

```v
//m_struct_test.v
import m

struct TestStruct {
	field map[m.MyEnum]string
}

fn test_struct() {
	s := TestStruct{
		field: map{
			m.MyEnum.one: "one"
		}
	}
}
```

```v
//m_alias_test.v
import m

type TestEnumMap = map[m.MyEnum]string

fn test_alias() {
	a := TestEnumMap(map{
		m.MyEnum.one: "one"
	})
	assert a[m.MyEnum.one] == "one"
}
```

```v
//m_local_struct_test.v
import m

fn test_local_struct() {
	s := m.MyStruct{
		field: map{
			m.MyEnum.one: "one"
		}
	}
	assert s.field[m.MyEnum.one] == "one"
}
```

```v
//m_local_alias_test.v
import m

fn test_local_alias() {
	a := m.MyEnumMap(map{
		m.MyEnum.one: "one"
	})
	assert a[m.MyEnum.one] == "one"
}
```

**What did you expect to see?**

All tests pass.
 
**What did you see instead?**

Only `m_literal_test.v` and `m_local_struct_test.v` pass.

The other three tests fail with the following messages:

```
.\m_struct_test.v
m_struct_test.v:4:20: error: maps only support string, integer, float, rune, enum or voidptr keys for now (not `m.MyEnum`)
    2 |
    3 | struct TestStruct {
    4 |     field map[m.MyEnum]string
      |                       ^
    5 | }

.\m_alias_test.v
m_alias_test.v:3:32: error: maps only support string, integer, float, rune, enum or voidptr keys for now (not `m.MyEnum`)
    1 | import m
    2 |
    3 | type TestEnumMap = map[m.MyEnum]string

.\m_local_alias_test.v
.../Temp/v/test_session_1772700/m_local_alias_test.exe.1338615834930266346.tmp.c:418: warning: WINVER redefined
.../Temp/v/test_session_1772700/m_local_alias_test.exe.1338615834930266346.tmp.c:871: error: ';' expected (got "m__MyEnumMap")
builder error:
==================
C error. This should never happen.
```

PR https://github.com/vlang/v/pull/10234 will fix `m_alias_test.v`, `m_struct_test.v`

The last test still fails with 
```
/tmp/v/testy.14929796013817314708.tmp.c:756: error: ';' expected (got "m__MyEnumMap")
...
```

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -770,27 +770,6 @@ pub fn (mut g Gen) write_typedef_types() {
 			continue
 		}
 		match typ.kind {
-			.alias {
-				parent := unsafe { &g.table.type_symbols[typ.parent_idx] }
-				is_c_parent := parent.name.len > 2 && parent.name[0] == `C` && parent.name[1] == `.`
-				mut is_typedef := false
-				if parent.info is ast.Struct {
-					is_typedef = parent.info.is_typedef
-				}
-				mut parent_styp := parent.cname
-				if is_c_parent {
-					if !is_typedef {
-						parent_styp = 'struct ' + parent.cname[3..]
-					} else {
-						parent_styp = parent.cname[3..]
-					}
-				} else {
-					if typ.info is ast.Alias {
-						parent_styp = g.typ(typ.info.parent_type)
-					}
-				}
-				g.type_definitions.writeln('typedef $parent_styp $typ.cname;')
-			}
 			.array {
 				g.type_definitions.writeln('typedef array $typ.cname;')
 			}
@@ -851,7 +830,11 @@ static inline void __${typ.cname}_pushval($typ.cname ch, $el_stype val) {
 			}
 		}
 	}
-
+	for typ in g.table.type_symbols {
+		if typ.kind == .alias && typ.name !in c.builtins {
+			g.write_alias_typesymbol_declaration(typ)
+		}
+	}
 	// Generating interfaces after all the common types have been defined
 	// to prevent generating interface struct before definition of field types
 	for typ in g.table.type_symbols {
@@ -859,6 +842,28 @@ static inline void __${typ.cname}_pushval($typ.cname ch, $el_stype val) {
 			g.write_interface_typesymbol_declaration(typ)
 		}
 	}
+}
+
+pub fn (mut g Gen) write_alias_typesymbol_declaration(sym ast.TypeSymbol) {
+	parent := unsafe { &g.table.type_symbols[sym.parent_idx] }
+	is_c_parent := parent.name.len > 2 && parent.name[0] == `C` && parent.name[1] == `.`
+	mut is_typedef := false
+	if parent.info is ast.Struct {
+		is_typedef = parent.info.is_typedef
+	}
+	mut parent_styp := parent.cname
+	if is_c_parent {
+		if !is_typedef {
+			parent_styp = 'struct ' + parent.cname[3..]
+		} else {
+			parent_styp = parent.cname[3..]
+		}
+	} else {
+		if sym.info is ast.Alias {
+			parent_styp = g.typ(sym.info.parent_type)
+		}
+	}
+	g.type_definitions.writeln('typedef $parent_styp $sym.cname;')
 }
 
 pub fn (mut g Gen) write_interface_typesymbol_declaration(sym ast.TypeSymbol) {

--- a/vlib/v/tests/alias_custom_type_map_test.v
+++ b/vlib/v/tests/alias_custom_type_map_test.v
@@ -2,7 +2,7 @@ import geometry
 
 fn test_aliases_map_init() {
 	a := geometry.ShapeMap(map{
-		geometry.Shape.circle: "Shape is a circle."
+		geometry.Shape.circle: 'Shape is a circle.'
 	})
-	assert a[geometry.Shape.circle] == "Shape is a circle."
+	assert a[geometry.Shape.circle] == 'Shape is a circle.'
 }

--- a/vlib/v/tests/alias_custom_type_map_test.v
+++ b/vlib/v/tests/alias_custom_type_map_test.v
@@ -1,0 +1,8 @@
+import geometry
+
+fn test_aliases_map_init() {
+	a := geometry.ShapeMap(map{
+		geometry.Shape.circle: "Shape is a circle."
+	})
+	assert a[geometry.Shape.circle] == "Shape is a circle."
+}

--- a/vlib/v/tests/modules/geometry/geometry.v
+++ b/vlib/v/tests/modules/geometry/geometry.v
@@ -10,6 +10,8 @@ pub enum Shape {
 	triangle
 }
 
+pub type ShapeMap = map[Shape]string
+
 // used by vlib/v/tests/map_enum_keys_test.v
 pub enum Form3D {
 	sphere

